akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSourceSpec.scala

```diff
@@ -33,6 +33,22 @@ class QueueSourceSpec extends StreamSpec {
 
   "A QueueSource" must {
 
+    "return a QueueClosed event if queue is completed" in {
+      val source = Source.queue[String](0, OverflowStrategy.backpressure)
+      val q1 = source.to(Sink.ignore).run()
+      q1.complete()
+      Await.ready(q1.watchCompletion(), pause)
```

Comment:
currently only awaiting on the future would trigger the bug inside the test case, `futureValue`, would basically run to fast and still enqueue the element and thus resulted in a always green test.
However awaiting the future and calling q1.offer would timeout the test, since the promise would never return.